src/dhcpmon/src/dhcp_device.c

```diff
@@ -242,62 +222,158 @@ static void read_callback(int fd, short event, void *arg)
 }
 
 /**
- * @code dhcp_device_validate(counters, counters_snapshot);
+ * @code dhcp_device_is_dhcp_inactive(counters);
  *
- * @brief validate current interface counters by comparing aggregate counter with snapshot counters.
+ * @brief Check if there were no DHCP activity
+ *
+ * @param counters  current/snapshot counter
+ *
+ * @return true if there were no DHCP activity, false otherwise
+ */
+static bool dhcp_device_is_dhcp_inactive(uint64_t counters[][DHCP_DIR_COUNT][DHCP_MESSAGE_TYPE_COUNT])
+{
+    uint64_t *rx_counters = counters[DHCP_COUNTERS_CURRENT][DHCP_RX];
+    uint64_t *rx_coutner_snapshot = counters[DHCP_COUNTERS_SNAPSHOT][DHCP_RX];
+
+    bool rv = true;
+    for (uint8_t i = 0; (i < monitored_msg_sz) && rv; i++) {
+        rv = rx_counters[monitored_msgs[i]] == rx_coutner_snapshot[monitored_msgs[i]];
```

Comment:
rv = rv && (...) ?

Reply:
It is the save as assigning as start condition is true and we stop at first failure.

`rv = true && (..)` -> `rv = (..)` Please note the AND'ing in the for loop construct:
```
    for (uint8_t i = 0; (i < monitored_msg_sz) && rv; i++) {
```